src/eve/visitors.py

```diff
@@ -33,11 +33,16 @@
     Iterable,
     MutableSequence,
     MutableSet,
+    Optional,
     Tuple,
     Union,
+    cast,
 )
 
 
+PREVISITOR = Callable[[concepts.TreeNode, Dict[str, Any]], Dict[str, Any]]
```

Comment:
I have two comments here: 
- I think a type alias should have a `CamelCased` name like classes. For example:

```suggestion
PrevisitCallable = Callable[[concepts.TreeNode, Dict[str, Any]], Dict[str, Any]]
```
- Wouldn't it be more coherent if Previsitors receive the usual unpacked `kwargs` instead of a dict? For typing, this can be expressed using a simple callback `Protocol` (https://www.python.org/dev/peps/pep-0544/#callback-protocols)